ui/signed-out/src/commonMain/kotlin/social/androiddev/signedout/navigation/DefaultSelectServerComponent.kt

```diff
@@ -10,13 +10,54 @@
 package social.androiddev.signedout.navigation
 
 import com.arkivanov.decompose.ComponentContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import social.androiddev.domain.authentication.usecase.AuthenticateClient
 
 class DefaultSelectServerComponent(
     private val componentContext: ComponentContext,
+    // TODO Remove passing up server since we will be storing it and can retrieve from storage
+    // in view model
     private val launchOAuth: (server: String) -> Unit,
-) : SelectServerComponent, ComponentContext by componentContext {
+) : KoinComponent, SelectServerComponent, ComponentContext by componentContext {
+
+    private val authenticateClient: AuthenticateClient by inject()
+
+    // TODO: There is support through Decompose for proper state handling
+    // and adding lifecycle aware components. We should hook into this so we can
+    // properly cancel the coroutineScope (similar to viewModelScope extension)
+    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+    private val _state = MutableStateFlow(SelectServerComponent.State())
+    override val state: StateFlow<SelectServerComponent.State> = _state
 
     override fun onServerSelected(server: String) {
```

Comment:
Im not terribly familiar with Koin but this feels like a class that is doing too much. Is it a component that owns dependencies, a state store or somewhere we will have and test business logic?

Reply:
Yeah so these component classes are a convention from Decompose. They essentially hold the business and navigation logic. 

I will probably move some of the business logic and specific view state to a "view model" instead, and keep just the navigation logic within these components.